test/integration/controller_test.go

```diff
@@ -1315,64 +1032,28 @@ func TestServiceBindingOrphanMitigation(t *testing.T) {
 	}
 }
 
-// TestBasicFlowsAsyncBinding tests the same flows as TestBasicFlowsAsync, using
-// asynchronous binding/unbindg as well.
-func TestBasicFlowsAsyncBinding(t *testing.T) {
-	// Enable the AsyncBindingOperations feature
-	utilfeature.DefaultFeatureGate.Set(fmt.Sprintf("%v=true", scfeatures.AsyncBindingOperations))
-	defer utilfeature.DefaultFeatureGate.Set(fmt.Sprintf("%v=false", scfeatures.AsyncBindingOperations))
-
-	_, catalogClient, _, _, _, _, shutdownServer, shutdownController := newTestController(t, fakeosb.FakeClientConfiguration{
-		CatalogReaction: &fakeosb.CatalogReaction{
-			Response: getTestCatalogResponse(),
-		},
-		ProvisionReaction: &fakeosb.ProvisionReaction{
-			Response: &osb.ProvisionResponse{
-				Async: true,
-			},
-		},
-		PollLastOperationReaction: &fakeosb.PollLastOperationReaction{
-			Response: &osb.LastOperationResponse{
-				State: osb.StateSucceeded,
-			},
-		},
-		PollBindingLastOperationReaction: &fakeosb.PollBindingLastOperationReaction{
-			Response: &osb.LastOperationResponse{
-				State: osb.StateSucceeded,
-			},
-		},
-		UpdateInstanceReaction: &fakeosb.UpdateInstanceReaction{
-			Response: &osb.UpdateInstanceResponse{
-				Async: true,
-			},
-		},
-		BindReaction: &fakeosb.BindReaction{
-			Response: &osb.BindResponse{
-				Async: true,
-			},
-		},
-		GetBindingReaction: &fakeosb.GetBindingReaction{
-			Response: &osb.GetBindingResponse{
-				Credentials: map[string]interface{}{
-					"foo": "bar",
-					"baz": "zap",
-				},
-			},
-		},
-		UnbindReaction: &fakeosb.UnbindReaction{
-			Response: &osb.UnbindResponse{
-				Async: true,
-			},
-		},
-		DeprovisionReaction: &fakeosb.DeprovisionReaction{
-			Response: &osb.DeprovisionResponse{
-				Async: true,
-			},
-		},
-	})
+// TestAsyncProvisionWithMultiplePolls tests an async instance provisioning
+// where the first few last-operation polls respond still in-progress.
+func TestAsyncProvisionWithMultiplePolls(t *testing.T) {
+	_, catalogClient, _, osbClient, _, _, shutdownServer, shutdownController := newTestController(t)
 	defer shutdownController()
 	defer shutdownServer()
 
+	osbClient.ProvisionReaction.Response.Async = true
+
+	const NumberOfInProgressResponses = 2
+	numberOfPolls := 0
+	osbClient.AddHook(PreHook, HookOperationPollLastOperation, func(request interface{}, response interface{}, err error) {
```

Comment:
wow I did not know go scope worked like this!

Reply:
The anonymous function declaration as an argument to AddHook?

Reply:
Or osbClient being available inside of it?